### PR TITLE
complete post-comment

### PR DESCRIPTION
**Pull request by Smasheroonie**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -174,3 +174,112 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    };
+    return request(app)
+      .post("/api/articles/dogs/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+});
```
Comment (nc-stephen-corke) [inline __tests__/app.test.js:285]: Testing is very thorough here including consideration of some edge cases. Tests are well written and easy to follow.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -174,3 +174,112 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      username: "rogersop",
+      body: "You should probably shorten this article",
+    };
+
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body).toMatchObject({
```
Comment (nc-stephen-corke) [inline __tests__/app.test.js:190]: It would be better for the comment to be on an appropriate key rather than just being the response body itself

File: controllers/comments-controller.js [hunk 3]
```diff
@@ -12,3 +12,14 @@ exports.getComments = (req, res, next) => {
...
     .catch(next);
 };
+
+exports.addComment = (req, res, next) => {
+  const { username, body } = req.body;
+  const { article_id } = req.params;
+
+  postComment(username, body, article_id)
+    .then((comment) => {
+      res.status(201).send(comment);
```
Comment (nc-stephen-corke) [inline controllers/comments-controller.js:22]: To make this endpoint extendable you should return the data on a key

File: controllers/errors-controller.js [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
+  switch (err.code) {
+    case "22P02":
+      res.status(400).send({ msg: "Bad request" });
+      break;
+    case "23503":
+      res.status(404).send({ msg: "Not found" });
+      break;
+    default:
+      next(err);
+      break;
```
Comment (nc-stephen-corke) [inline controllers/errors-controller.js:11]: Nice use of a `switch` here

File: models/comments-model.js [hunk 5]
```diff
@@ -1,11 +1,34 @@
...
+  }
+
+  let queryStr = `
+    INSERT INTO comments(author, body, article_id) VALUES ($1, $2, $3) RETURNING *`;
+
+  return db.query(queryStr, [username, body, article_id]).then(({ rows }) => {
+    const comment = rows[0];
+    return comment;
+  });
+};
```
Comment (nc-stephen-corke) [inline models/comments-model.js:34]: Some good customer error handling here but also you've demonstrated a good understanding of which errors Postgres will throw.

**Review by nc-stephen-corke — COMMENTED** [review]
Nice work, very thorough and great solutions. Worth adding a key to the repsonse.
